MTA_limit_trends/Comp/run_comp_grad_data_update.py

```diff
@@ -224,32 +239,37 @@ def update_comp_data(gname, msid_list, eyear, etime):
 #
 #--- set the input fits file name
 #
-                fits = sub_dir + gname + '/' + msid + '_full_data_' + str(year) + '.fits'
+                fits = f"{sub_dir}/{gname}/{msid}_full_data_{year}.fits"
                 if not os.path.isfile(fits):
-                    fits = sub_dir +  gname + '/' + msid + '_full_data_' + str(year) + '.fits.gz'
+                    fits = f"{fits}.gz"
                     if not os.path.isfile(fits):
+                        print(f"Could not find {fits[:-3]} or gz.")
                         continue
 #
 #--- extract the data part needed and save in a fits file 
 #
-                out = extract_data_from_deposit(msid, fits, tstart, etime, dtype, alimit)
-                if out == False:
+                tbhdu = extract_data_from_deposit(msid, fits, tstart, etime, dtype, alimit)
+                if tbhdu == False:
                     print("Something went wrong for " + msid + ' in year: ' + str(year))
                     continue
 
                 else:
+                    #If not false, then managed to create the table HDU,
+                    #save to a fits file and append to original table
+                    appendfile = f"{DATA_DIR}/{gname}/{msid}_{dtype}_append.fits"
+                    tbhdu.writeto(appendfile)
```

Comment:
Part of restructure with changing the create_fits_file function into a create_fits_table function. See comment on this function for explanation.